processor/transformprocessor/internal/traces/registry.go

```diff
@@ -0,0 +1,84 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traces // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/traces"
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+)
+
+var registry = make(map[string]interface{})
+
+func newFunctionCall(inv common.Invocation) (exprFunc, error) {
+	if f, ok := registry[inv.Function]; ok {
+		fType := reflect.TypeOf(f)
+		args := make([]reflect.Value, 0)
+		for i := 0; i < fType.NumIn(); i++ {
+			argType := fType.In(i)
```

Comment:
Can we avoid whole this complicated reflection logic by providing a factory for the exprFunc that is in the registry? Then we pass all arguments to the factory?

Reply:
The motivation for this was to make writing functions as easy as possible - after the framework settles, we'll be writing many more functions than the framework. I think the factory you are suggesting would need to receive the `common.Invocation`, which would complicate the function more than being able to just declare what it wants (getSetter, getter, string, etc) and let the framework resolve it vs the user config

Reply:
I don't want to block this, but would like to think more about this. We can have a TODO/issue or something to think about this. I have a feeling that there may be a solution to avoid this, otherwise I agree we should not complicate the user in general.